server/utils/secrets/secret_ref_utils.py

```
@@ -363,8 +362,8 @@ def _clear_secret_ref(self, user_id: str, provider: str) -> None:
             set_rls_context(cursor, conn, user_id, log_prefix="[SecretRef:clearRef]")
             cursor.execute(
                 f"UPDATE user_tokens SET is_active = FALSE, secret_ref = '' "
-                f"WHERE user_id = %s AND provider = %s {clause}",
-                (user_id, provider) + params,
+                f"WHERE {predicate} AND provider = %s",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)